openslides/utils/collection.py

```diff
@@ -117,14 +161,236 @@ def get_full_data(self):
+        In other cases use raw=True to add the same cache key.
+        """
+        key = get_element_list_cache_key(self.collection_string)
+        if raw:
+            key = self.make_key(key)
+        return key
+
+    def get_model(self):
+        """
+        Returns the django model that is used for this collection.
+        """
+        return get_model_from_collection_string(self.collection_string)
+
+    def element_generator(self):
+        """
+        Generator that yields all collection_elements of this collection.
+        """
+        # Get all cache keys.
+        ids = self.get_all_ids()
+        cache_keys = [
+            get_single_element_cache_key(self.collection_string, id)
+            for id in ids]
+        cached_full_data_dict = cache.get_many(cache_keys)
+
+        # Get all ids that are missing.
+        missing_ids = set()
+        for cache_key in cache_keys:
+            if cache_key not in cached_full_data_dict:
+                missing_ids.add(get_collection_id_from_cache_key(cache_key)[1])
+
+        # Generate collection elements that where in the cache.
+        for cache_key, cached_full_data in cached_full_data_dict.items():
+            yield CollectionElement.from_values(
+                *get_collection_id_from_cache_key(cache_key),
+                full_data=cached_full_data)
+
+        # Generate collection element that where not in the cache.
+        if missing_ids:
+            from openslides.core.config import config
+            if self.collection_string == config.get_collection_string():
+                # If config elements are not in the cache, they have to be read from
+                # the config object.
+                for key, value in config.items():
+                    if key in missing_ids:
+                        collection_element = CollectionElement.from_values(
+                            config.get_collection_string(),
+                            key,
+                            full_data={'key': key, 'value': value})
+                        # We can not use .from_instance therefore the config value
+                        # is not saved to the cache. We have to do it manualy.
+                        collection_element.save_to_cache()
+                        yield collection_element
+            else:
+                model = self.get_model()
+                try:
+                    query = model.objects.get_full_queryset()
+                except AttributeError:
+                    query = model.objects
+                for instance in query.filter(pk__in=missing_ids):
+                    yield CollectionElement.from_instance(instance)
+
+    def as_autoupdate_for_projector(self):
+        """
+        Returns a list of dictonaries to send them to the projector.
+        """
+        output = []
+        for collection_element in self.element_generator():
+            content = collection_element.as_autoupdate_for_projector()
```

Comment:
Here content may be None, right? Then we to not want `None` to be appended to the output.